internal/runners/activate/activate.go

```diff
@@ -49,7 +53,28 @@ func (r *Activate) Run(params *ActivateParams) error {
 }
 
 func (r *Activate) run(params *ActivateParams, activatorLoop activationLoopFunc) error {
-	logging.Debug("Activate %v, %v", params.Namespace, params.PreferredPath)
+	logging.Debug("Activate with namespace=%v, path=%v", params.Namespace, params.PreferredPath)
+
+	activeProjectDir := os.Getenv(constants.ActivatedStateEnvVarName)
+	alreadyActivated := activeProjectDir != ""
+
+	// handle case, if we are already activated
+	if alreadyActivated {
+		if params.Default && params.Namespace.IsValid() {
+			return locale.NewError("err_default_with_name_already_active", "Trying to set {{.V0}} as default project while in activated environment.  Please de-activate the current runtime.", params.Namespace.String())
+		}
+		if !params.Default {
```

Comment:
Why would this only happen if not using Default?

Reply:
This addresses the above AC:  If you run `state activate --default` it is never an error - even if already activated.  Otherwise, the code below will create the error message.